pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -62,6 +62,32 @@ def test_agg_ser_multi_key(df):
     tm.assert_series_equal(results, expected)
 
 
+def test_agg_different_partials():
+    # for issue 28570
+
+    def add_top_column(df, top_col, inplace=False):
```

Comment:
Looks like this function is only used once. Could you in-line the logic directly in the test?

Reply:
Rather, this _entire function_ is only used once so all the logic can be in-lined.